src/ReflectionProperty.php

```diff
@@ -75,12 +78,144 @@ public function __construct(
         unset($this->name, $this->class);
     }
 
+    /**
+     * Return string representation of this little old property.
+     *
+     * @return string
+     */
+    public function __toString(): string
+    {
+        $modifiers = implode(' ', BaseReflection::getModifierNames($this->getModifiers()));
+        $name      = $this->getName();
+        $default   = null;
+        if ($this->isDefault()) {
+            $quoteString = null;
+            $checkType   = true;
+
+            $default          = $this->getDefaultValue();
+            $defaultValueType = gettype($default);
+            $defaultNode      = $this->getNode()->default;
+            $declaringClass   = $this->getDeclaringClass();
+            $namespaceName    = $declaringClass->getNamespaceName();
+
+            // PHP >= 8.1 has changed the way how default values are represented
+            if (PHP_VERSION_ID >= 80100) {
```

Comment:
This entire logic of parsing/analyzing default value should be only in one single place - `NodeExpressionResolver` inside `ValueResolver` namespace. It was created to work on top of AST + given context (eg class or method).

There can be same places for `ReflectionParameter`, `ReflectionClassConstant`, etc. This is why it will be better use AST tree + context in all places.

If you have troubles only with `__toString()` methods in all classes - let's skip them, as in reality such methods are not used very often. You can mark them as skipped for now to avoid spending a lot of time on preparing same output - this might be time-consuming process

Reply:
I see, starting with that gave me a good look on the AST tree and how everything works.